sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotShiftByZeroOrIntSize.CSharp9.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+
+// Native Integers can be either 32b or 64b depending on the underlying system.
+// We should only raise when the shift is done with values >= 64.
```

Comment:
I think we should not raise for big values, because we don't know what's the `nint` size of the run machine. Will be 128 one day. Could be less on some super exotic scenario 🤯

Reply:
So do you suggest to avoid raising on `nint`? I'm fine with that

Reply:
Yes. Let's don't raise on `nint` and `nuint` even for `1 << 512`. 

I did it the same way in CBDE overflow rule. It's even more visible on that example.